firedrake/adjoint/ensemble_reduced_functional.py

```diff
@@ -50,20 +53,23 @@ class EnsembleReducedFunctional(ReducedFunctional):
     works, please refer to the `Firedrake manual
     <https://www.firedrakeproject.org/parallelism.html#id8>`_.
     """
-    def __init__(self, J, control, ensemble):
+    def __init__(self, J, control, ensemble, scatter_control=True):
         super(EnsembleReducedFunctional, self).__init__(J, control)
         self.ensemble = ensemble
+        self.scatter_control = scatter_control
 
     def __call__(self, values):
         local_functional = super(EnsembleReducedFunctional, self).__call__(values)
-        if isinstance(local_functional, float):
-            total_functional = self.ensemble.ensemble_comm.allreduce(sendobj=local_functional, op=MPI.SUM)
-        elif isinstance(local_functional, firedrake.Function):
-            total_functional = type(local_functional)(local_functional.function_space())
-            total_functional = self.ensemble.allreduce(local_functional, total_functional)
-        else:
-            raise NotImplementedError("This type of functional is not supported.")
-        return total_functional
+        if self.scatter_control:
```

Comment:
This is the same problem as before. Literally just get rid of this if. `self.scatter_control` should have no influence on anything that happens in `__call__`